samples/communication-between-nodes/src/main/java/com/badoo/ribs/samples/comms_nodes/rib/language_selector/builder/LanguageSelectorBuilder.kt

```diff
@@ -0,0 +1,39 @@
+package com.badoo.ribs.samples.comms_nodes.rib.language_selector.builder
+
+import com.badoo.ribs.builder.SimpleBuilder
+import com.badoo.ribs.core.modality.BuildParams
+import com.badoo.ribs.samples.comms_nodes.app.Language
+import com.badoo.ribs.samples.comms_nodes.rib.language_selector.LanguageSelector
+import com.badoo.ribs.samples.comms_nodes.rib.language_selector.LanguageSelectorNode
+import com.badoo.ribs.samples.comms_nodes.rib.language_selector.LanguageSelectorPresenter
+import com.badoo.ribs.samples.comms_nodes.rib.language_selector.LanguageSelectorPresenterImpl
+import com.badoo.ribs.samples.comms_nodes.rib.language_selector.LanguageSelectorView
+
+class LanguageSelectorBuilder(
+    private val languages: List<Language>,
+    private val currentLanguage: Language
+) : SimpleBuilder<LanguageSelector>() {
+
+    override fun build(buildParams: BuildParams<Nothing?>): LanguageSelector {
+        val presenter = LanguageSelectorPresenterImpl(languages = languages, defaultSelection = getCurrentSelection())
+        val viewDependency = object : LanguageSelectorView.Dependency {
+            override val presenter: LanguageSelectorPresenter
+                get() = presenter
+        }
+
+        return LanguageSelectorNode(
+            buildParams = buildParams,
+            viewFactory = LanguageSelector.Customisation().viewFactory.invoke(viewDependency),
```

Comment:
Never create `Customisations` on the fly, it makes it impossible to override them. `ViewFactory` instance should be grabbed in build method from current instance of `Customisation` coming from `buildParams`. You can take a look at any other Builder in sandbox app.

```suggestion
            viewFactory = viewFactory.invoke(viewDependency),
```